borg/archiver.py

```diff
@@ -286,13 +292,25 @@ def do_extract(self, args):
         manifest, key = Manifest.load(repository)
         archive = Archive(repository, key, manifest, args.location.archive,
                           numeric_owner=args.numeric_owner)
-        patterns = adjust_patterns(args.paths, args.excludes)
+
+        matcher = PatternMatcher()
+        if args.excludes:
+            matcher.add(args.excludes, False)
```

Comment:
maybe that could be a `__init__` param excludes=... and the if-check could then happen in `__init__`.

same for includes.